fhir-notification-nats/src/main/java/com/ibm/fhir/notifications/nats/impl/FHIRNotificationNATSPublisher.java

```diff
@@ -0,0 +1,228 @@
+            service.subscribe(this);
+            log.info("Initialized NATS publisher for channel '" + channelName + "' using servers: '" + servers + "'.");
+        } catch (Throwable t) {
+            String msg = "Caught exception while initializing NATS publisher.";
+            log.log(Level.SEVERE, msg, t);
+            throw new IllegalStateException(msg, t);
+        } finally {
+            log.exiting(this.getClass().getName(), "init");
+        }
+    }
+
+    /**
+     * Performs any necessary "shutdown" logic to disconnect from the channel.
+     */
+    public void shutdown() {
+        log.entering(this.getClass().getName(), "shutdown");
+
+        try {
+            if (log.isLoggable(Level.FINE)) {   
+                log.fine("Shutting down NATS publisher for channel: '" + channelName + "'.");
+            }
+            if (sc != null) {
+               sc.close();
+            }
+        } catch (Throwable t) {
+            String msg = "Caught exception shutting down NATS publisher for channel: '" + channelName + "'.";
+            log.log(Level.SEVERE, msg, t);
+            throw new IllegalStateException(msg, t);
+        } finally {
+            log.exiting(this.getClass().getName(), "shutdown");
+        }
+    }
+
+    /**
+     * Publishes an event to NATS.
+     */
+    @Override
+    public void notify(FHIRNotificationEvent event) throws FHIRNotificationException {
+        log.entering(this.getClass().getName(), "notify");
+        String jsonString = null;
+        try {
+            jsonString = FHIRNotificationUtil.toJsonString(event, true);
+
+            if (log.isLoggable(Level.FINE)) { 
+                log.fine("Publishing NATS notification event to channel '" + channelName + "',\nmessage: '" + jsonString + "'.");
+            }
+            
+            sc.publish("FHIRNotificationEvent", jsonString.getBytes(), acb);
+    
+            if (log.isLoggable(Level.FINE)) {
+                log.fine("Published NATS notification event to channel '" + channelName + "'");
+            }
+        } catch (Throwable e) {
+            String msg = buildNotificationErrorMessage(channelName, (jsonString == null ? "<null>" : jsonString));
+            log.log(Level.SEVERE, msg , e);
+            throw new FHIRNotificationException(msg, e);
+        } finally {
+            log.exiting(this.getClass().getName(), "notify");
+        }
+    }
+    
+    /**
+     * Builds a formatted error message to indicate a notification publication failure.
+     */
+    private String buildNotificationErrorMessage(String channelName, String notificationEvent) {
+        return String.format("NATS publication failure; channel '%s'\nNotification event: '%s'\n.", channelName, notificationEvent);
+    }
+}
+
+/* 
+* Modified from original NATS documentation @ https://docs.nats.io/developing-with-nats/security/tls
+*
+* This example requires certificates to be in the java keystore format (.jks).
```

Comment:
Its funny because we literally just changed all our other keystores from JKS to PKCS12.
Can NATS work with PKCS12 or it really needs JKS?

Reply:
The .jks keystore and truststore files we create are of type pkcs12, so I believe we could just change the name to .p12.  That should be easy enough to test.